torch/csrc/jit/jit_opt_bisect.cpp

```diff
@@ -0,0 +1,121 @@
+#include <cstdlib>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <ATen/core/function.h>
+#include <c10/util/Exception.h>
+#include <c10/util/StringUtil.h>
+#include <torch/csrc/jit/api/function_impl.h>
+#include <torch/csrc/jit/frontend/error_report.h>
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/jit_opt_bisect.h>
+#include <torch/csrc/jit/serialization/python_print.h>
+
+namespace torch {
+namespace jit {
+
+static int parseOptLimit(const std::string& opt_limit) {
+  // TODO bad code
+  try {
+    int64_t n = std::stoi(opt_limit);
+    return n;
+  } catch (...) {
+    return -1;
+  }
+}
+
+static std::unordered_map<std::string, int64_t> parseJITBisectOption(
+    const char* option) {
+  std::stringstream in_ss;
+  in_ss << "function:";
+  if (option) {
+    in_ss << option;
+  }
+
+  std::unordered_map<std::string, int64_t> passes_to_opt_limits;
+  std::string line;
+  while (std::getline(in_ss, line, ':')) {
+    if (line.size() == 0) {
+      continue;
+    }
+    auto index_at = line.find_last_of('=');
+    auto pass_name = line.substr(0, index_at);
+    pass_name = c10::detail::ExcludeFileExtension(pass_name);
+    auto opt_limit = parseOptLimit(line.substr(index_at + 1));
+    passes_to_opt_limits.insert({pass_name, opt_limit});
+  }
+
+  return passes_to_opt_limits;
+}
+
+bool is_bisect_enabled(const char* pass_name) {
+  static const char* opt_limit = std::getenv("PYTORCH_OPT_LIMIT");
+  // if nothing is provided, let's allow everything
+  if (!opt_limit) {
+    return true;
```

Comment:
Nit: I find the name of the function a bit confusing: it says "bisect is enabled", but when the `PYTORCH_OPT_LIMIT ` flag is not set, then we return `true`.